soh/src/overlays/actors/ovl_En_Tk/z_en_tk.c

```diff
@@ -502,8 +502,12 @@ void EnTk_Init(Actor* thisx, PlayState* play) {
 
     CollisionCheck_SetInfo2(&this->actor.colChkInfo, NULL, &sColChkInfoInit);
 
-    if (gSaveContext.dayTime <= 0xC000 || gSaveContext.dayTime >= 0xE000 || !!LINK_IS_ADULT ||
-        play->sceneNum != SCENE_SPOT02) {
+    if (CVarGetInteger("gDampeAllNight", 0)) {
+        if (gSaveContext.dayTime <= (int)-1 || !!LINK_IS_ADULT || play->sceneNum != SCENE_SPOT02) {
```

Comment:
`gSaveContext.dayTime` is a `uint16_t` so comparing against a negative number isn't right in this case.

There is the `IS_NIGHT` and `IS_DAY` macros that is a more clear check.

But if Dampe only ever spawns in the night scene, then you probably don't even need to check for night at all.

Reply:
> `gSaveContext.dayTime` is a `uint16_t` so comparing against a negative number isn't right in this case.
> 
> There is the `IS_NIGHT` and `IS_DAY` macros that is a more clear check.
> 
> But if Dampe only ever spawns in the night scene, then you probably don't even need to check for night at all.

You're right, I don't even need the time check.